Momentum Changes + new card experiment

CardboardBones opened the pull request:
- Changed Momentum to be cut in half at the end of your turn
- Updated CAR and ALTERNATOR to only increase their damage equal to your current momentum
- Airstrafe is now a 1 cost skill (needs new upgrade)
- Changed Sprint so that it's upgrade no longer provides more momentum
- added Rush/Mastiff cards. I have not been able to test them yet, but they don't crash the game

CardboardBones committed:
- Changed Momentum to be cut in half at the end of your turn
- Updated CAR and ALTERNATOR to only increase their damage equal to your current momentum
- Airstrafe is now a 1 cost skill (needs new upgrade)
- Changed Sprint so that it's upgrade no longer provides more momentum
- added Rush/Mastiff cards. I have not been able to test them yet, but they don't crash the game

Cartopol commented on pom.xml:
```diff
@@ -18,17 +18,17 @@
             See README.md (the "How to build from source" section) for details.
         -->
-        <steamappsPath>${env.STEAMAPPS_PATH}</steamappsPath>
+        <steamappsPath>"/Users/micah/Library/Application Support/Steam/steamapps"</steamappsPath>
```
Unfortunately, if I merge this it will no longer work for me. I'll have to experiment with how I can leave this file out of the merge.

Cartopol commented on src/main/java/pilot/cards/pilot/Alternator.java:
```diff
@@ -22,23 +23,37 @@ public class Alternator extends CustomPilotModCard {
         }
         this.addToBot(new AttackDamageRandomEnemyAction(this, AttackEffect.SLASH_HORIZONTAL));
         this.addToBot(new AttackDamageRandomEnemyAction(this, AttackEffect.SLASH_HORIZONTAL));
+        this.baseDamage -= p.getPower(MomentumPower.POWER_ID).amount;
```
Using the calculateBonusBaseDamage() above, the entire use method can be simplified to: 
```
this.addToBot(new AttackDamageRandomEnemyAction(this, AttackEffect.SLASH_HORIZONTAL));
      this.addToBot(new AttackDamageRandomEnemyAction(this, AttackEffect.SLASH_HORIZONTAL));
```
My bad, I had forgotten about the calculateBonusBaseDamage() when I initially made this card.

CardboardBones replied:
Sweet, yeah I wasn't quite sure what was happening here. Essentially, I noticed that the card was keeping the scaling with your momentum, meaning that if you increased it by 5 and then drew it on a later turn, it would still be increased by that 5 on top of your current momentum. I decided to just add the bit that subtracted by your current momentum. I see what I need to do here.

Cartopol commented on src/main/java/pilot/cards/pilot/CAR.java:
```diff
@@ -51,6 +51,9 @@ public void use(AbstractPlayer p, AbstractMonster m) {
         addToBot(new DamageAction(m, new DamageInfo(p, damage), AttackEffect.SLASH_HORIZONTAL));
+       if (((Pilot)p).hasMomentum() && p.hasPower(MomentumPower.POWER_ID)) {
+           this.baseDamage -= p.getPower(MomentumPower.POWER_ID).amount;
+       }
```
This should not be necessary. I believe it does nothing, since the bonusBaseDamage is calculated continuously.

CardboardBones replied:
I'll need to delete the "this.baseDamage += p.getPower(MomentumPower.POWER_ID).amount;" as well

Cartopol commented on src/main/java/pilot/cards/pilot/Rush.java:
```diff
@@ -0,0 +1,42 @@
+    @Override
+    public void use(AbstractPlayer p, AbstractMonster m) {
+        addToBot(new MakeTempCardInHandAction(this.cardsToPreview.makeStatEquivalentCopy(), 1));
+        addToBot(new DiscardAction(p, p, magicNumber, false));
```
If you don't want to give the option of discarding the mastiff, you could flip these around.

CardboardBones replied:
great point, thank you

Cartopol commented on src/main/java/pilot/cards/pilot/tempCards/Mastiff.java:
```diff
@@ -0,0 +1,55 @@
+public class Mastiff extends CustomPilotModCard {
+    public static final String ID = PilotMod.makeID(Mastiff.class);
+
+    private static final CardRarity RARITY = CardRarity.RARE;
```
This should have CardRarity.SPECIAL to avoid it coming up in draft picks.

CardboardBones replied:
ahhhh, that's what it was!

Cartopol commented on src/main/java/pilot/cards/pilot/tempCards/Mastiff.java:
```diff
@@ -0,0 +1,55 @@
+
+    public void upgrade() {
+        if (!upgraded){
+            upgradeDamage(UPGRADED_PLUS_DMG);
```
You'll also want to add upgradeName() and upgradeDescription to this method, otherwise the upgrade will not be shown on the card.

Cartopol commented on src/main/java/pilot/powers/MomentumPower.java:
```diff
@@ -32,8 +32,14 @@ public float modifyBlock(float blockAmount) {
+            }
+
+            if (this.amount > 1) {
+                int reduceBy = this.amount / 2;
```
This could be simplified quite a bit. First off, there's some redundant code to clean up. The only thing your if-branch needs to do, is determine the reduceBy value, all the other bits inside the scope of the if-statement is the same. Once you know the reduceBy, you only need to have one call to the ReducePowerAction and the logger. 

Another thing to note, is that there's a neat way to write very succinct if-then-else statements, which would look a bit like this:

`amount > 1 ? reduceBy = amount / 2 : reduceBy = 1`

To break that down a bit, it basically reads: Is amount > 1 ? Then reduceBy = amount. Else, reduceBy = 1. The question mark signals this as an if statement, and the colon starts the else clause.

Cartopol replied:
Also, this change should be described in PilotMod-Power-String.json, so that the tooltip corresponds with the mechanics.

CardboardBones replied:
thanks for breaking down the code lingo, I'll definitely change it to this. Ill get the tooltip updated as well

Cartopol commented on src/main/resources/pilotResources/localization/eng/PilotMod-Card-Strings.json:
```diff
@@ -25,6 +25,10 @@
+  "pilot:Mozambique": {
+    "NAME": "Mozambique",
+    "DESCRIPTION": "Reflex. NL Deal !D! damage and apply !M! Vulnerable."
+  },
```
This is currently unused. Is this a new card you're working on? No problem leaving it in here for now.

CardboardBones replied:
It was an old card from a while back that I ended up deleting later. I just had forgotten to delete this bit as well

Cartopol commented on src/main/resources/pilotResources/localization/eng/PilotMod-Card-Strings.json:
```diff
@@ -135,5 +139,13 @@
+  },
+  "pilot:Mastiff": {
+    "NAME": "Mastiff",
+    "DESCRIPTION": " Deal !D! Damage and gain !M! Momentum. NL Advantage: Double the damage dealt."
```
The card currently doesn't double the damage, it adds a second damage instance. If you want it to double the damage, you can add a calculateBonusBaseDamage() call, in which you have something like 
`if player.hasAdvantage() {
return baseDamage; 
}`
I believe that should do the thing.

CardboardBones replied:
Oh nice! Yeah I initially wanted it to double but i couldn't figure it out, so I just quickly changed it to attack twice. I noticed that later when testing after uploading this.

Cartopol reviewed (changes requested):
Overall a super solid first set of changes! You're definitely doing a lot of things right, and looking at the right things for examples. A couple of things to fix, but nothing major.

CardboardBones committed:
- Changed Momentum to be cut in half at the end of your turn
- Updated CAR and ALTERNATOR to only increase their damage equal to your current momentum
- Airstrafe is now a 1 cost skill (needs new upgrade)
- Changed Sprint so that it's upgrade no longer provides more momentum
- added Rush/Mastiff cards.

CardboardBones committed:
Delete pom.xml

Cartopol committed:
Merge branch 'master' into MICAH_-_Momentum/modified_cards
